src/FGS.Linq.Extensions.OneOf.EntityFramework.Shared/OneOfQueryableExtensions.cs

```diff
@@ -0,0 +1,77 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+#if EF6
+using System.Data.Entity;
+#endif
+using FGS.Collections.Extensions.OneOf.Units;
+using FGS.Linq.Extensions.EntityFramework;
+#if EFCORE
+using Microsoft.EntityFrameworkCore;
+#endif
+using OneOf;
+
+namespace FGS.Linq.Extensions.OneOf.EntityFramework
+{
+    /// <summary>
+    /// Extends <see cref="IQueryable{T}"/> with a materialization methods that describe their failure conditions using discriminated unions.
+    /// </summary>
+    public static class OneOfQueryableExtensions
+    {
+        /// <summary>
+        /// Asynchronously returns a single element from the query represented by <paramref name="source"/>, returns <see cref="NoElements"/> if there are no elements, returns <see cref="MoreThanOneElement"/> if there is more than one element.
+        /// </summary>
+        /// <typeparam name="T">The type of items that <paramref name="source"/> queries for.</typeparam>
+        /// <param name="source">The source query.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> to observe while waiting for the materialization to complete.</param>
+        /// <returns>A task that represents the asynchronous operation. The task result contains the single element from the input query, or the reason why it could be returned.</returns>
+        public static async Task<OneOf<T, NoElements, MoreThanOneElement>> MaterializeSingleOrReasonWhyNot<T>(this IQueryable<T> source, CancellationToken cancellationToken = default)
+        {
+            var firstOrDefault = (await source.Take(1).MaterializeAsync(cancellationToken).ConfigureAwait(false)).First();
```

Comment:
Should this be `...FirstOrDefault()`? Or perhaps just `await source.MarterializeFirstOrDefaultAsync....` as you have done in the other methods in this class?

Reply:
I'm gonna go with `FirstOrDefault()`.

While writing this, I had an implicit bias against using the `Materialize…OrDefault` methods because I'm tempted to eventually move those into a separate package from `Materialize` so-as to discourage their use (in favor of these, instead.)

Reply:
✅

Reply:
✅